cpp/util/read_key.cc

```diff
@@ -1,8 +1,10 @@
 #include "util/read_key.h"
 
-#include <openssl/pem.h>
-#include <openssl/engine.h>
+#ifndef OPENSSL_IS_BORINGSSL
 #include <openssl/conf.h>
+#include <openssl/engine.h>
+#endif
```

Comment:
Could you make the `#endif`s be `#endif  // OPENSSL_IS_BORINGSSL`? 
It makes it easier to spot the extent.